Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -1,130 +1,188 @@
-﻿//
+//
 // Copyright ical.net project maintainers and contributors.
 // Licensed under the MIT license.
 //
 
-using System;
-using System.IO;
+#nullable enable
 using Ical.Net.Serialization.DataTypes;
 using Ical.Net.Utility;
 using NodaTime;
+using System;
+using System.Globalization;
+using System.IO;
 
 namespace Ical.Net.DataTypes;
 
 /// <summary>
 /// The iCalendar equivalent of the .NET <see cref="DateTime"/> class.
 /// <remarks>
 /// In addition to the features of the <see cref="DateTime"/> class, the <see cref="CalDateTime"/>
-/// class handles time zone differences, and integrates seamlessly into the iCalendar framework.
+/// class handles timezones, and integrates seamlessly into the iCalendar framework.
 /// </remarks>
 /// </summary>
 public sealed class CalDateTime : EncodableDataType, IDateTime
 {
-    public static CalDateTime Now => new CalDateTime(DateTime.Now);
+    // The date part that is used to return the Value property.
+    private DateOnly _dateOnly;
+    // The time part that is used to return the Value property.
+    private TimeOnly _timeOnly;
 
-    public static CalDateTime Today => new CalDateTime(DateTime.Today);
+    private const double AlmostZeroEpsilon = 1e-10;
+    private const string UtcTzId = "UTC";
+
+    /// <summary>
+    /// Gets the current date/time in the local timezone.
+    /// </summary>
+    public static CalDateTime Now => new CalDateTime(DateTime.Now, null, true);
```

Comment:
If we want to offer a `Now`, then this is one of the very few places where referring to the system's local time would make sense.